src/test/java/com/google/sps/SurveyServletAndResponseTest.java

```diff
@@ -0,0 +1,413 @@
+ * gcloud config set project [PROJECT_NAME]
+ *
+ * Skeleton code for Datastore emulator from:
+ * https://stackoverflow.com/questions/40348653/google-datastore-emulator-using-java-not-using-gae
+ */
+public class SurveyServletAndResponseTest {
+
+    protected static LocalDatastoreHelper localDatastoreHelper;
+    protected Datastore datastore;
+    protected KeyFactory keyFactory;
+
+    @BeforeClass
+    public static void setUpClass() throws InterruptedException, IOException {
+        // Create and start a local datastore emulator on a random free port
+        System.out.println("[Datastore-Emulator] start");
+        localDatastoreHelper = LocalDatastoreHelper.create();
+        localDatastoreHelper.start();
+        System.out.println("[Datastore-Emulator] listening on port: " + localDatastoreHelper.getPort());
+
+        // Set the system property to tell the gcloud lib to use the datastore emulator
+        System.setProperty("DATASTORE_EMULATOR_HOST","localhost:" + localDatastoreHelper.getPort());
+    }
+
+    @Before
+    public void setUp() {
+        // Create the datastore instance
+        // Because of the system property set it in setUpClass() this
+        // datastore will be connected with the datastore emulator.
+        datastore = DatastoreOptions.getDefaultInstance().getService();
+        keyFactory = datastore.newKeyFactory().setProjectId("manage-at-scale-step-2020").setKind("TestEntity");
+    }
+
+    @After
+    public void tearDown() throws IOException {
+        System.out.println("[Datastore-Emulator] reset");
+        // This resets the datastore after every test
+        localDatastoreHelper.reset();
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws InterruptedException, IOException, TimeoutException {
+        System.out.println("[Datastore-Emulator] stop");
+        // This stops the datastore emulator after all tests are done
+        localDatastoreHelper.stop();
+    }
+
+    @Test
+    public void test1() {
+        // Stores an entity in the datastore and retrieves it later
+
+        // Create an Entity "TestEntity"
+        Entity.Builder builder = Entity.newBuilder(keyFactory.newKey(42));
+        builder.set("name", "Test1");
+
+        // Store it in datastore
+        datastore.put(builder.build());
+
+        // Retrieve entity by key
+        Entity entity = datastore.get(keyFactory.newKey(42));
+        assertNotNull(entity);
+        assertEquals("Test1", entity.getString("name"));
+    }
+
+    @Test
+    public void test2() {
+        // Try to access the entity created in test1, shouldn't work because
+        // of calling reset in tearDown() after each test.
+
+        // Try to retrieve entity by key
+        Entity entity = datastore.get(keyFactory.newKey(42));
+        assertNull(entity);
+    }
+
+    
+    // Creating SurveyResponse instances to use in tests:
+
+    private static ImmutableMap<PanasFeelings, PanasIntensity> fooFeelings;
```

Comment:
nit: put all instance variables at the top of the class (where the others ones currently are)

Reply:
Done in b7c691a7a57e17973d25f1384368e9c642e38641